transformer_engine/common/CMakeLists.txt

```
@@ -14,6 +14,21 @@ set(CMAKE_CUDA_STANDARD_REQUIRED ON)
 
 project(transformer_engine LANGUAGES CUDA CXX)
 
+if(NOT DEFINED ENV{NVTE_MAX_BUILD_THREADS})
+  set(ENV{NVTE_MAX_BUILD_THREADS} "1")
+endif()
+set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} --threads $ENV{NVTE_MAX_BUILD_THREADS}")
```

I'd prefer if we didn't affect the environment:
```suggestion
set(BUILD_THREADS_PER_JOB $ENV{NVTE_BUILD_THREADS_PER_JOB})
if (NOT BUILD_THREADS_PER_JOB)
  set(BUILD_THREADS_PER_JOB 1)
endif()
set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} --threads ${BUILD_THREADS_PER_JOB}")
```

---

This `ENV{}` only sets env for cmake, not the env in the shell. I don't see an issue with that.